File: tests/fixtures/kotlin/RealWorld.kt
package com.example.test

/**
 * A sealed interface representing a result.
 */
sealed interface Result<out T> {
    fun isSuccess(): Boolean
}

/**
 * Represents a successful result.
 */
data class Success<T>(val value: T) : Result<T> {
    override fun isSuccess(): Boolean = true
}

/**
 * Represents a failed result.
 */
data class Failure(val error: Throwable) : Result<Nothing> {
    override fun isSuccess(): Boolean = false
}

/**
 * Direction enum class.
 */
enum class Direction {
    NORTH, SOUTH, EAST, WEST;

    fun isHorizontal(): Boolean = this == EAST || this == WEST
}

/**
 * Color enum class with properties.
 */
enum class Color(val rgb: Int) {
    RED(0xFF0000),
    GREEN(0x00FF00),
    BLUE(0x0000FF);

    fun containsRed(): Boolean = (rgb and 0xFF0000) != 0
}

/**
 * A regular interface.
 */
interface Repository<T> {
    suspend fun findById(id: String): T?
    suspend fun findAll(): List<T>
    suspend fun save(entity: T): T
    suspend fun delete(id: String)
}

/**
 * Abstract class.
 */
abstract class BaseViewModel {
    abstract fun onStart()
    open fun onStop() {}
}

/**
 * Companion object and nested class.
 */
class NetworkClient private constructor(val baseUrl: String) {

    companion object {
        fun create(baseUrl: String): NetworkClient = NetworkClient(baseUrl)
    }

    class Builder {
        private var baseUrl: String = ""

        fun baseUrl(url: String): Builder {
            baseUrl = url
            return this
        }

        fun build(): NetworkClient = NetworkClient(baseUrl)
    }
}

/**
 * Object declaration (singleton).
 */
object Logger {
    fun log(message: String) {
        println(message)
    }
}

/**
 * Top-level functions.
 */
fun <T> retry(times: Int, block: () -> T): T {
    repeat(times - 1) {
        try {
            return block()
        } catch (e: Exception) {
            // retry
        }
    }
    return block()
}

suspend fun fetchData(url: String): String {
    return "data from $url"
}

/**
 * Extension functions.
 */
fun String.isValidEmail(): Boolean {
    return this.contains("@") && this.contains(".")
}

fun List<Int>.secondOrNull(): Int? = if (size >= 2) this[1] else null

/**
 * Type alias.
 */
typealias StringMap = Map<String, String>

/**
 * Annotation class.
 */
annotation class Inject

/**
 * Inline class / value class.
 */
@JvmInline
value class Email(val value: String)

File: tests/fixtures/swift/RealWorld.swift
import Foundation

/// A protocol defining repository operations.
protocol Repository {
    associatedtype Entity
    func findById(_ id: String) -> Entity?
    func findAll() -> [Entity]
    func save(_ entity: Entity) -> Entity
    func delete(_ id: String)
}

/// A simple struct.
struct Point {
    let x: Double
    let y: Double

    func distance(to other: Point) -> Double {
        let dx = x - other.x
        let dy = y - other.y
        return (dx * dx + dy * dy).squareRoot()
    }
}

/// A class with inheritance.
class Animal {
    let name: String

    init(name: String) {
        self.name = name
    }

    func speak() -> String {
        return ""
    }
}

/// Subclass.
class Dog: Animal {
    override func speak() -> String {
        return "Woof!"
    }
}

/// An enum with associated values.
enum Result<Success, Failure> {
    case success(Success)
    case failure(Failure)

    var isSuccess: Bool {
        switch self {
        case .success: return true
        case .failure: return false
        }
    }
}

/// A simple enum.
enum Direction {
    case north, south, east, west

    var isHorizontal: Bool {
        return self == .east || self == .west
    }
}

/// An enum with raw values.
enum HTTPStatusCode: Int {
    case ok = 200
    case notFound = 404
    case serverError = 500
}

/// Extensions.
extension String {
    func isValidEmail() -> Bool {
        return contains("@") && contains(".")
    }

    var isBlank: Bool {
        return trimmingCharacters(in: .whitespaces).isEmpty
    }
}

extension Array where Element: Comparable {
    func secondSmallest() -> Element? {
        guard count >= 2 else { return nil }
        return sorted()[1]
    }
}

/// Protocol with default implementation via extension.
protocol Drawable {
    func draw()
}

extension Drawable {
    func draw() {
        print("Default drawing")
    }
}

/// A generic class.
class NetworkManager<T: Codable> {
    let baseURL: URL

    init(baseURL: URL) {
        self.baseURL = baseURL
    }

    func fetch(endpoint: String) -> T? {
        return nil
    }
}

/// Actor (Swift 5.5+).
actor BankAccount {
    private var balance: Double

    init(balance: Double) {
        self.balance = balance
    }

    func deposit(_ amount: Double) {
        balance += amount
    }

    func withdraw(_ amount: Double) -> Bool {
        guard balance >= amount else { return false }
        balance -= amount
        return true
    }
}

/// Top-level function.
func retry<T>(times: Int, block: () throws -> T) rethrows -> T {
    for _ in 0..<(times - 1) {
        do { return try block() } catch { continue }
    }
    return try block()
}

/// A typealias.
typealias Completion = (Result<String, Error>) -> Void

/// Property wrapper.
@propertyWrapper
struct Clamped<Value: Comparable> {
    var wrappedValue: Value {
        didSet { wrappedValue = min(max(wrappedValue, range.lowerBound), range.upperBound) }
    }
    let range: ClosedRange<Value>

    init(wrappedValue: Value, _ range: ClosedRange<Value>) {
        self.range = range
        self.wrappedValue = min(max(wrappedValue, range.lowerBound), range.upperBound)
    }
}
